src/compiler/checker.ts

```diff
@@ -10514,6 +10519,9 @@ namespace ts {
                     return indexedAccess;
                 }
             }
+            if (isGenericTupleType(type.objectType)) {
+                return getIndexTypeOfType(type.objectType, IndexKind.Number);

```

Comment:
Yeah, we should only do this when the index type is number-like.

Reply:
So, currently we error on indexed access types with variadic tuple types for any string-like key because we need to do more reasoning about constrains of variadic tuple types. That means you can't actually observe this issue. I think it is fine to leave this as is and address post-beta.